### PR TITLE
Add Alda configuration layer

### PR DESCRIPTION
**Pull request by astroengisci**
Alda is a music-composition language which has an Emacs major mode but not a Spacemacs layer.

### CODE REVIEW

File: layers/+lang/alda/packages.el [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+               alda-play-buffer)
+    :init
+    (progn
+      (spacemacs/set-leader-keys-for-major-mode 'alda-mode
+        "r" 'alda-play-region
+        "c" 'alda-play-block
+        "n" 'alda-play-line
+        "b" 'alda-play-buffer))))
+
+(defun alda/post-init-ggtags ()
```
Comment (d12frosted) [inline layers/+lang/alda/packages.el]: You need to define `ggtags` in the `alda-packages` variable.

**Comment by astroengisci** [comment]
Thanks, @d12frosted, I have made the requested fixes. The issue with `ggtags` was just that I misunderstood the `post-init-` functions, so I renamed it to `post-init-alda-mode`.

File: layers/+lang/alda/README.org [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   
+* Description
+Alda is a music composition language allowing music to easily be written and
+edited in a text file.
+
+This layer adds keybindings for alda-mode's functions, which allow Alda code
+to be interpreted and played by the running Alda server.
+
+** Features:
+    - Syntax highlighting for Alda files.
```
Comment (d12frosted) [inline layers/+lang/alda/README.org]: There should be no spaces here.

File: layers/+lang/alda/README.org [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ - [[#description][Description]]
+ - [[#install][Install]]
+ - [[#key-bindings][Key bindings]]
+   
+* Description
+Alda is a music composition language allowing music to easily be written and
+edited in a text file.
+
+This layer adds keybindings for alda-mode's functions, which allow Alda code
+to be interpreted and played by the running Alda server.
```
Comment (d12frosted) [inline layers/+lang/alda/README.org]: Just a question - does one need to manually run Alda server in order to play the music?

File: layers/+lang/alda/packages.el [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+               alda-play-block
+               alda-play-line
+               alda-play-buffer)
+    :init
+    (progn
+      (spacemacs/set-leader-keys-for-major-mode 'alda-mode
+        "r" 'alda-play-region
+        "c" 'alda-play-block
+        "n" 'alda-play-line
+         "b" 'alda-play-buffer))))
```
Comment (d12frosted) [inline layers/+lang/alda/packages.el]: Sorry for the nit-picking, but there is an extra space 😛

File: layers/+lang/alda/packages.el [hunk 5]
```diff
@@ -0,0 +1,30 @@
+;;; packages.el --- Alda Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2017 Sylvain Benner & Contributors
```
Comment (d12frosted) [inline layers/+lang/alda/packages.el]: The year is 2018 now 😸 It's a common 'issue': https://github.com/syl20bnr/spacemacs/pull/11440

File: layers/+lang/alda/README.org (hunk 3 above)
Comment (astroengisci) [reply layers/+lang/alda/README.org]: `alda-mode` will automatically start an Alda server if one is not running, and the user will have to run the command again. I'll update the README to reflect this.

**Comment by sdwolfz** [comment]
@astroengisci if you squash all your commits into one then I'll go ahead and merge this layer.

**Comment by astroengisci** [comment]
@sdwolfz Sorry I didn't work on this until now, Thanksgiving Break is coming up and everything is due all at once.

Not sure why the documentation check is failing, what do I need to do to resolve that?

**Comment by duianto** [comment]
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.

The documentation check failing was probably just a test server issue.

**Pull request closed without merging**